src/tokenized-buffer.coffee

```diff
@@ -126,9 +125,17 @@ class TokenizedBuffer extends Model
     @disposables.add(@configSubscriptions)
 
     @retokenizeLines()
-    @packageManager.triggerActivationHook("#{grammar.packageName}:grammar-used")
+
     @emitter.emit 'did-change-grammar', grammar
 
+    # Delay this to the next tick to ensure whoever created the buffer has the
+    # chance to listen for this event before we send it.
+    process.nextTick =>
```

Comment:
I am not sure how I feel about using `process.nextTick` here, because it seems like the pattern we're trying to prevent here is that the users of this API write something like:

```js
let buffer = new TokenizedBuffer
this.handleGrammarUsed(buffer.getGrammar())
buffer.onDidChangeGrammar(this.handleGrammarUsed)
```

Which isn't necessarily wrong or bad, but I get that we could have a nicer API for this. Similar to your proposal above, how would you feel about using the `observeFoo` pattern, as we do [elsewhere in the codebase](https://github.com/atom/atom/blob/master/src/workspace.coffee#L255-L262)?

```js
let buffer = new TokenizedBuffer
buffer.observeGrammar(this.handleGrammarUsed)
```